registrations/metrics.py

```diff
@@ -233,6 +271,90 @@ def registrations_change_messaging_total_last(self, start, end):
             .filter(action='change_messaging')\
             .count()
 
+    def optout_msg_type_sum(self, msg_type, start, end):
+        result = utils.search_optouts({
+            "created_at__gt": start,
+            "created_at__lte": end,
+        })
+
+        identities = []
+        for data in result:
+            identities.append(data['identity'])
```

Comment:
We can probably do the same `set(data['identity'] for data in result)` that we did for the metrics firing here in the metrics regeneration (for all the metric types here).